source/docs/software/advanced-control/trajectories/troubleshooting.rst

```diff
@@ -0,0 +1,246 @@
+Troubleshooting
+===============
+
+Troubleshooting complete failures
+---------------------------------
+There are a number of things that can cause your robot to do completely the wrong thing. The below checklist covers some common mistakes.
+
+* My robot doesn't move.
+
+    - Are you actually outputting to your motors?
+    - Is a ``MalformedSplineException`` getting printed to the driver station? If yes, go to the ``MalformedSplineException`` section below.
+    - Is your trajectory very short or in the wrong units?
+* My robot swings around to drive the trajectory facing the other direction.
+
+    - Are the start and end headings of your trajectory wrong?
+    - Is your robot's gyro getting reset to the wrong heading?
+    - :ref:`Do you have the reverse flag set incorrectly? <docs/software/advanced-control/trajectories/trajectory-generation:Creating the trajectory config>`
+    - Are your gyro angles clockwise positive? If so, you should negate them.
+* My robot just drives in a straight line even though it should turn.
+
+    - Is your gyro set up correctly and returning good data?
+    - Are you passing your gyro heading to your odometry object with the correct units?
+    - Is your track width correct? Is it in the correct units?
+* I get a ``MalformedSplineException`` printout on the driver station and the robot doesn't move.
+
+    - :ref:`Do you have the reverse flag set incorrectly? <docs/software/advanced-control/trajectories/trajectory-generation:Creating the trajectory config>`
+    - Do you have two waypoints very close together with approximately opposite headings?
+    - Do you have two waypoints with the same (or nearly the same) coordinates?
+* My robot drives way too far.
+
+    - Are your encoder unit conversions set up correctly?
+    - Are your encoders connected?
+* My robot mostly does the right thing, but it's a little inaccurate.
+
+    - Go to the next section.
+
+Troubleshooting poor performance
+--------------------------------
+
+.. note:: This section is mostly concerned with troubleshooting poor trajectory tracking performance like a meter of error, not catastrophic failures like compilation errors, robots turning around and going in the wrong direction, or ``MalformedSplineException``\s.
```

Comment:
What's the \s for?

Reply:
If I remove the `\` then I get "Inline literal start-string without end-string." I could add a space, but then you get a space after the code block.